kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/TopicNameBasedKekSelector.java

```diff
@@ -21,8 +21,9 @@ public abstract class TopicNameBasedKekSelector<K> {
     /**
      * Returns a completion stage whose value, on successful completion, is a map from each of the given topic
      * names to the KEK id to use for encrypting records in that topic.
+     *
      * @param topicNames A set of topic names
      * @return A completion stage for the map form topic name to KEK id.
      */
-    public abstract @NonNull CompletionStage<Map<String, K>> selectKek(@NonNull Set<String> topicNames);
-}
+    public abstract @NonNull CompletionStage<Map<String, EncryptionScheme<K>>> selectKek(@NonNull Set<String> topicNames);
```

Comment:
I think this really means the class/interface should be renamed from `KekSelector` to `EncryptionSchemeSelector` or something else which implies more than just selecting Kek's.